quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -1388,8 +1380,15 @@ fn tags_filter_expression_helper(tags: &TagFilterAst) -> Cond {
 /// Builds a SQL query that returns indexes which match at least one pattern in
 /// `index_id_patterns`. For each pattern, we check if the pattern is valid and replace `*` by `%`
 /// to build a SQL `LIKE` query.
-fn build_index_id_patterns_sql_query(index_id_patterns: Vec<String>) -> anyhow::Result<String> {
-    assert!(!index_id_patterns.is_empty());
+fn build_index_id_patterns_sql_query(index_id_patterns: &[String]) -> anyhow::Result<String> {
+    if index_id_patterns.is_empty() {
+        anyhow::bail!("The list of index id patterns may not be empty.");
+    }
+    if index_id_patterns == ["*"] {
+        return Ok("SELECT * FROM indexes".to_string());
+    }
+    // TODO Fix me.
```

Comment:
Is the `TODO Fix me.` and the following comment are outdated? 
The check is handled in the `root_search` function and not at the metastore level.

Reply:
yes thx